installscripts/wizard/scenarios/support/jazz_jenkins.py

```diff
@@ -124,18 +129,20 @@ def get_and_add_docker_jenkins_config(jenkins_docker_path):
     """
         Launch a dockerized Jenkins server.
     """
-    os.chdir(jenkins_docker_path)
-    print("Running docker launch script")
-    subprocess.call([
-        'sg', 'docker', './launchscript.sh', '|', 'tee', '-a',
-        '../../docker_creation.out'
-    ])
-    # Get values to create the array
-    parameter_list = []
-    with open("docker_jenkins_vars") as f:
-        for line in f:
-            parameter_list.append(line.rstrip())
+    print "\nPlease provide Subnets details, since ECS Fargate will be configured with ALB"
+    jenkins_subnet = raw_input("Please enter the subnetid 1 :")
+    jenkins_subnet2 = raw_input("Please enter the subnetid 2 :")
```

Comment:
For ECS fargates , ALB is using for services so, for ALB we need atleast two subnets. Installer instance has only one subnet.

Reply:
Are we creating them? We must be, so why are we asking the user for them?

Reply:
We are creating the ALB only, using the subnets provided by the user.

Reply:
We need to talk about this. If we are expecting the user to create subnets manually and provide that info to us, we need to talk about why that is necessary and how to document it.

If we are expecting them to give us their existing subnet info, I would suggest we try to detect their current subnet info for them, rather than ask people to type them in.

Reply:
Yeah, basically an instance is attached to only one subnet. ie, from ec2-metadata we get only a single subnet. Also, You must specify subnets from at least two Availability Zones for creating the Application load balancer.

Reply:
Can we not create those 2 subnets ourselves with Terraform, and assign them to the ALB?

Reply:
@bleggett It will be difficult to handle the CIDR block input while creating the subnet. Especially the network settings. Right now, we are assuming the same vpc will be the user wish to configure in ECS fargate also. Other way is, we need to prepopulate the vpc, subnets and security group and ask the user for confirmation.

Reply:
We can't just ask for a VPC ID and create our own subnets and SGs?

Reply:
@bleggett In that case, we might ned to ask vpc id and internet gawatewayid. Ref: https://github.com/cloudposse/terraform-aws-dynamic-subnets

Reply:
@rajeevr2715 We can get the VPCs' CIDR block at runtime with either boto3 or `awscli`, but to your point, we need a _subnet_ CIDR block this 3rd-party module can make use of to spin off subnets. I am OK with asking for all 3 of those things in the installer, since we are then asking for things the user already has created (probably), and not asking them to create new things outside of our installer.

However, what I would prefer to do is create our own Jazz VPC in terraform and then just stand everything up there with terraform. If we start needing specific CIDR blocks and want to avoid collisions with existing stuff we need to be doing this anyway, that's what VPCs are for, with the addition of this ECS stuff we're starting to get too large and complex on the AWS side to be sharing a VPC with other stuff. That also means we wouldn't need this third-party Terraform module.

@suryajak @devsatishm Thoughts?

Reply:
> @rajeevr2715 We can get the current VPCs' CIDR block at runtime with either boto3 or `awscli`, but to your point, we need a nonoverlapping _subnet_ CIDR block range this 3rd-party module can make use of to spin off subnets. I am OK with asking for all 3 of those things in the installer (igwid, vpcid, subnet CIDR block), since we are then at least asking for things the user has already created (probably), and not asking them to manually create new AWS resources outside of our installer, like before.
> 
> However, what I would prefer to do is create our own Jazz VPC in terraform and then just stand everything up there with terraform. If we start needing specific CIDR blocks and want to avoid collisions with existing stuff we need to be doing this anyway, that's what VPCs are for, with the addition of this ECS stuff we're starting to get too large and complex on the AWS side to be sharing a VPC with other things. Doing that also means we wouldn't need this third-party Terraform module.
> 
> @suryajak @devsatishm Thoughts?

@bleggett @rajeevr2715  I believe we can go with two options like we did for other parts of jazz infrastructure. Like below - 

- `create new vpc` as the default option so that Jazz creates a new VPC, IGW (attaches to the VPC), two subnets within the VPC (using /24?). Admin only provides CIDR block for the VPC creation.
- `use existing vpc` as the second option so that Jazz uses an existing VPC. Admin provides `vpc-id` & Jazz creates two subnets within the VPC (using /24?)

Reply:
@devsatishm @bleggett Implemented the default (create new vpc, subnets) and tested scenario3.

Reply:
Implemented the existing vpc scenario also.